File: 240516/xor 결과 최대 만들기/max-of-xor.py
n, m = map(int, input().split())
arr = list(map(int, input().split()))

def xor(arr):
    start = arr[0]
    for i in range(1, len(arr)):
        start = start ^ arr[i]
    return start

ans = 0
def back(cnt, sm):
    global ans

    if cnt >= n:
        if len(sm) == m:
            ans = max(ans, xor(sm))
        return

    back(cnt+1, sm+[arr[cnt]])
    back(cnt+1, sm)

back(0, [])
print(ans)

File: 260501/XOR 결과 최대 만들기/max-of-xor.java
import java.util.*;

public class Main {
    public static int n, m;
    public static int[] A;
    public static int ans;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        m = sc.nextInt();
        A = new int[n];
        for (int i = 0; i < n; i++) {
            A[i] = sc.nextInt();
        }
        back(0, 0, 0);
        System.out.print(ans);

    }

    public static void back(int currIdx, int cnt, int result)
    {
        if (currIdx >= n)
        {
            if (cnt == m)
            {
                xor(result);
            }
            return;
        }
        back(currIdx+1, cnt+1, result ^ A[currIdx]);
        back(currIdx+1, cnt, result);
    }

    public static void xor(int result)
    {
        ans = Math.max(ans, result);
    }
}